packages/provider/src/KnownRelaysManager.ts

```diff
@@ -143,11 +149,39 @@ export class KnownRelaysManager {
     return relayEventParts.flat()
   }
 
-  async _fetchRecentlyActiveRelayManagers (): Promise<Set<Address>> {
+  async _getLatestRelayHubEvents (eventNames?: string[], managers?: Set<Address>): Promise<EventData[]> {
     const toBlock = await this.contractInteractor.getBlockNumber()
-    const fromBlock = Math.max(0, toBlock - this.config.relayLookupWindowBlocks)
 
-    const relayEvents: any[] = await this.getPastEventsForHub(fromBlock, toBlock)
+    let relayEvents: EventData[]
+    if (toBlock - this.latestScannedBlock < 10) {
+      // re-use the same cached events for ~2 minutes.
+      relayEvents = this.cachedLatestEvents
+    } else {
+      const fromBlock = Math.max(0, toBlock - this.config.relayLookupWindowBlocks)
```

Comment:
So as I was suggesting, registration lookup window should be separate (and few times bigger) than any event lookup window